idea/src/org/jetbrains/kotlin/idea/inspections/ReplaceWithEnumMapInspection.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.caches.resolve.resolveImportReference
+import org.jetbrains.kotlin.idea.project.platform
+import org.jetbrains.kotlin.idea.util.ImportInsertHelper
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtPsiFactory
+import org.jetbrains.kotlin.psi.callExpressionVisitor
+import org.jetbrains.kotlin.psi.createExpressionByPattern
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameUnsafe
+import org.jetbrains.kotlin.resolve.jvm.platform.JvmPlatform
+import org.jetbrains.kotlin.types.typeUtil.isEnum
+
+private val hashMapCreationFqNames = setOf(
+    "java.util.HashMap.<init>",
+    "kotlin.collections.HashMap.<init>",
+    "kotlin.collections.hashMapOf"
```

Comment:
What about `mapOf` and `linkedMapOf`, as well as `LinkedHashMap.<init>`?

Reply:
These cases are more difficult, as `mapOf(Enum.A to 5, Enum.B to 7)` does not generate nice code with `EnumMap`, it would probably become something like `EnumMap().let { it[Enum.A] = 5; it[Enum.B] = 7;` Furthermore, `mapOf` implies immutability, where `EnumMap` is mutable.

With `LinkedHashMap`, behavior is different: `LinkedHashMap` guarantees the maintence of insertion order, where EnumMap has natural enum order. 

These are the reasons for which I excluded these method calls. If it is not a problem, then I will at them anyways.

Reply:
Clear, I think we should leave it as is.